WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java

```diff
@@ -1,60 +1,110 @@
 package org.wordpress.android.ui.accounts;
 
 import android.os.Bundle;
+import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentTransaction;
 import android.support.v7.app.AppCompatActivity;
+import android.view.MenuItem;
 import android.widget.Toast;
 
 import org.wordpress.android.R;
 import org.wordpress.android.WordPress;
 import org.wordpress.android.ui.accounts.login.LogInOrSignUpFragment;
-import org.wordpress.android.ui.accounts.login.nav.LoginEvents;
-import org.wordpress.android.ui.accounts.login.nav.LoginNavigationController;
-import org.wordpress.android.ui.accounts.login.nav.LoginNavigationFsmGetter;
-import org.wordpress.android.ui.accounts.login.nav.LoginState;
+import org.wordpress.android.ui.accounts.login.LoginEmailFragment;
+import org.wordpress.android.ui.accounts.login.LoginNavFragment;
+import org.wordpress.android.ui.accounts.login.nav.LoginNavHandler;
+import org.wordpress.android.ui.accounts.login.nav.LoginStateGetter;
 
-public class LoginActivity extends AppCompatActivity implements
-        LoginNavigationFsmGetter,
-        LoginNavigationController.ContextImplementation {
-
-    LoginNavigationController mLoginNavigationController = new LoginNavigationController(LoginState.PROLOGUE, this);
+public class LoginActivity extends AppCompatActivity implements LoginNavHandler, LoginStateGetter.FsmGetter {
+    private static final String TAG_LOGIN_NAV_FRAGMENT = "TAG_LOGIN_NAV_FRAGMENT";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         ((WordPress) getApplication()).component().inject(this);
 
-        setContentView(R.layout.welcome_activity);
+        setContentView(R.layout.login_activity);
+
+        addLoginNavFragment();
 
         if (savedInstanceState == null) {
             addLoginPrologueFragment();
         }
     }
 
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            onBackPressed();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onBackPressed() {
+        // perform the Nav back first so updated fragments can find their expected state there.
+        retrieveLoginNavFragment().goBack();
```

Comment:
I revised the implementation (with 4841cb6) to make it a bit cleaner to use. The `getLoginNavFragment()` can still throw but, that will reveal a bug in our code: we shouldn't use the method until after the initial addition of the LoginNavFragment. No need to guard for null since this is not recoverable.